sqlx-core/src/migrate/migrate.rs

```diff
@@ -14,7 +14,7 @@ pub trait MigrateDatabase {
 
     // drop database in url
     // uses a maintenance database depending on driver
-    fn drop_database(url: &str) -> BoxFuture<'_, Result<(), Error>>;
+    fn drop_database(url: &str, _force: bool) -> BoxFuture<'_, Result<(), Error>>;
```

Comment:
This is unfortunately a breaking change which would have to wait for 0.8.0.

A backwards-compatible addition would be to add a new method with a default implementation:

```suggestion
    fn drop_database(url: &str) -> BoxFuture<'_, Result<(), Error>>;
    
    fn force_drop_database(url: &str) -> BoxFuture<'_, Result<(), Error>> {
        unimplemented!("Database driver does not support force-dropping a database")
    }
```

Instead of panicking, however, you could add a new variant to `MigrateError`, something like `ForceNotSupported`. That's backwards-compatible since the enum is `#[non_exhaustive]`.

Alternatively, it could just default to calling `.drop_database()` but I feel like if someone passed `--force` and it still failed, they might be confused and/or annoyed.